numpy/lib/_iotools.py

```diff
@@ -735,17 +735,17 @@ def upgrade(self, value):
         self._checked = True
         try:
             return self._strict_call(value)
-        except ValueError:
+        except ValueError as e:
             # Raise an exception if we locked the converter...
             if self._locked:
                 errmsg = "Converter is locked and cannot be upgraded"
-                raise ConverterLockError(errmsg)
+                raise ConverterLockError(errmsg) from e
             _statusmax = len(self._mapper)
             # Complains if we try to upgrade by the maximum
             _status = self._status
             if _status == _statusmax:
                 errmsg = "Could not find a valid conversion function"
-                raise ConverterError(errmsg)
+                raise ConverterError(errmsg) from e
```

Comment:
This looks like it is in the same situation as the previous one: it makes a noisy exception, but there might be useful info. in there, so OK.

Reply:
As above